state/protocol/kvstore.go

```diff
@@ -101,8 +108,104 @@ type KVStoreReader interface {
 	//    enough that the network is overwhelming likely to finalize at least one
 	//    block with a view in this range
 	GetFinalizationSafetyThreshold() uint64
+
+	// v2
+
+	// GetCadenceComponentVersion returns the Cadence component version.
+	// Error Returns:
+	//   - kvstore.ErrKeyNotSupported if invoked on a KVStore instance before v2.
+	//   - kvstore.ErrKeyNotSet if the key has no value
+	GetCadenceComponentVersion() (MagnitudeOfChangeVersion, error)
```

Comment:
What if we condense it and instead of returning version `GetCadenceComponentVersion` and `GetDacdenceComponentVersionUpgrade` we implement `GetCadenceComponentVersion` with return type `UpdatableField[MagnitudeOfChangeVersion]` ?

Reply:
I'm inclined to leave it as two separate functions. Most use-cases probably aren't dealing with pending updates, they just want to know the current value, so I feel like it's worth the extra function to make the typical query path simpler.